File: examples-sdk/go/402-agent-http-researcher/main.go
// 402-agent-http-researcher — Go SDK equivalent
// Compile: vil compile --from go --input 402-agent-http-researcher/main.go --release
package main

import vil "github.com/OceanOS-id/vil-go"

func main() {
	s := vil.NewServer("http-researcher-agent", 3121)

	research_agent := vil.NewService("research-agent")
	research_agent.Endpoint("POST", "/research", "research_handler")
	research_agent.Endpoint("GET", "/products", "products_handler")
	s.Service(research_agent)

	s.Compile()
}

File: examples-sdk/zig/402-agent-http-researcher/main.zig
// 402-agent-http-researcher — Zig SDK equivalent
// Compile: vil compile --from zig --input 402-agent-http-researcher/main.zig --release

const vil = @import("vil");

pub fn main() void {
    var server = vil.Server.init("http-researcher-agent", 3121);
    var research_agent = vil.Service.init("research-agent");
    research_agent.endpoint("POST", "/research", "research_handler");
    research_agent.endpoint("GET", "/products", "products_handler");
    server.service(&research_agent);
    server.compile();
}
